### PR TITLE
Settings: Invites: Integration

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

Hooks up the Invites dumb component to the real RPCs (API server endpoints) for listing, creating and deleting invitations.

There's a rendering change included too (thanks @chromakode): the settings nav wasn't setting `overflow: auto` on its children -- that, and putting `flex-shrink: 0` on every child component are required for cases like this one where the content doesn't fit the window vertically and we don't want to squash it.

It's pretty easy to test this in prod, since you can harmlessly create an invitation and then revoke it, all using this UI.

### CODE REVIEW

File: shared/actions/settings.js [hunk 1]
```diff
@@ -66,6 +93,135 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+    yield call(apiserverPostRpcPromise, {
+      param: {
+        endpoint: 'cancel_invitation',
+        args: [{key: 'invitation_id', value: inviteId}],
+      },
+    })
+    yield put(({
+      type: Constants.invitesReclaimed,
+      payload: undefined,
+      error: false,
```
Comment (chrisnojima) [inline shared/actions/settings.js]: shouldn't need `error: false`

File: shared/actions/settings.js [hunk 2]
```diff
@@ -66,6 +93,135 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+    })
+  } catch (err) {
+    // TODO hook into global error handler
+    console.error(err)
+  }
+}
+
+function * sendInviteSaga (invitesSendAction: InvitesSend): SagaGenerator<any, any> {
+  const {email, message} = invitesSendAction.payload
+  let args = [{key: 'email', value: email}]
```
Comment (chrisnojima) [inline shared/actions/settings.js]: could be const

File: shared/actions/settings.js [hunk 3]
```diff
@@ -66,6 +93,135 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+  if (message) {
+    args.push({key: 'invitation_message', value: message})
+  }
+  try {
+    yield call(apiserverPostRpcPromise, {
+      param: {
+        endpoint: 'send_invitation',
+        args: args,
```
Comment (chrisnojima) [inline shared/actions/settings.js]: could be just args

File: shared/actions/settings.js [hunk 4]
```diff
@@ -66,6 +93,135 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+      },
+    })
+    yield put(({
+      type: Constants.invitesSent,
+      payload: undefined,
+      error: false,
```
Comment (chrisnojima) [inline shared/actions/settings.js]: dont need error

**Comment by keybase-ci-visdiff** [comment]
The commits 57227776ad48...2cf6f2967d69 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Empty_Message-57227776ad48.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Empty_Message-2cf6f2967d69.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Empty-57227776ad48.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Empty-2cf6f2967d69.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Email_Error-57227776ad48.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal_Email_Error-2cf6f2967d69.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal-57227776ad48.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475601794565-57227776ad48-2cf6f2967d69-linux/Invites-Normal-2cf6f2967d69.png)  
   Invites-Normal comparison
</summary>

File: shared/actions/settings.js (hunk 1 above)
Comment (cjb) [inline shared/actions/settings.js]: > shouldn't need `error: false`

It's a bit of a common pattern:
```js
 λ git grep "error: false" | wc -l
14
```
I think I like it because `payload: undefined` looks unfriendly; it makes it not obvious through scanning that the code block is a success case.  If we could skip `payload: undefined` *too*, I'd feel better about skipping error.

What do you think?  If you don't like `error: false`, perhaps we should have a PR for removing it everywhere?

Comment (chrisnojima) [inline shared/actions/settings.js]: in my mind its all success case unless it has error=true. I'm fine with making payload optional (we follow FSA and it says it can be optional)

File: shared/actions/settings.js [hunk 5]
```diff
@@ -66,6 +89,135 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+
+function * refreshInvitesSaga (): SagaGenerator<any, any> {
+  try {
+    // If the rpc is fast don't clear it out first
+    const delayThenEmptyTask = yield fork(function * () {
+      yield call(delay, 500)
+      yield put({
+        type: Constants.invitesRefreshed,
+        payload: {
+          settings: null,
```
Comment (chrisnojima) [inline shared/actions/settings.js]: copy paste error?

**Comment by keybase-ci-visdiff** [comment]
The commits ad85056a69d2...85095d05b916 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Empty_Message-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Empty_Message-85095d05b916.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Empty-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Empty-85095d05b916.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Email_Error-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal_Email_Error-85095d05b916.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604407114-ad85056a69d2-85095d05b916-linux/Invites-Normal-85095d05b916.png)  
   Invites-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits ad85056a69d2...4bfafb768ce5 introduce visual changes on linux. <details><summary>:mag_right: 5 changed</summary>

 * Changed: **unlock_folders-Single** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/unlock_folders-Single-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/unlock_folders-Single-4bfafb768ce5.png)  
   unlock_folders-Single comparison
 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Empty_Message-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Empty_Message-4bfafb768ce5.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Empty-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Empty-4bfafb768ce5.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Email_Error-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal_Email_Error-4bfafb768ce5.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal-ad85056a69d2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475604579869-ad85056a69d2-4bfafb768ce5-linux/Invites-Normal-4bfafb768ce5.png)  
   Invites-Normal comparison
</summary>

File: shared/settings/dumb.js [hunk 6]
```diff
@@ -378,41 +378,38 @@ const invitesBase = {
...
       url: 'keybase.io/inv/9999999999',
     },
   ],
   acceptedInvites: [
     {
       id: '223456',
-      created: 1469565223000,
-      username: 'malgorithms',
+      created: 1469565223,
+      username: 'chris',
```
Comment (chromakode) [inline shared/settings/dumb.js]: :smirk:

File: shared/actions/settings.js (hunk 1 above)
Comment (cjb) [inline shared/actions/settings.js]: > I'm fine with making payload optional

Ah, looks like this isn't possible:

Marco: "Flow, I don’t think, has a notion of: If `P` is void, the key can be omitted."

**Comment by keybase-ci-visdiff** [comment]
The commits f666fb7a535b...4880aa49808c introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Empty_Message-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Empty_Message-4880aa49808c.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Empty-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Empty-4880aa49808c.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Email_Error-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal_Email_Error-4880aa49808c.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475616098336-f666fb7a535b-4880aa49808c-linux/Invites-Normal-4880aa49808c.png)  
   Invites-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits f666fb7a535b...e666105f8558 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Empty_Message-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Empty_Message-e666105f8558.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Empty-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Empty-e666105f8558.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Email_Error-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal_Email_Error-e666105f8558.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617630737-f666fb7a535b-e666105f8558-linux/Invites-Normal-e666105f8558.png)  
   Invites-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits f666fb7a535b...089d1f7b982d introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Empty_Message-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Empty_Message-089d1f7b982d.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Empty-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Empty-089d1f7b982d.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Email_Error-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal_Email_Error-089d1f7b982d.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475617757339-f666fb7a535b-089d1f7b982d-linux/Invites-Normal-089d1f7b982d.png)  
   Invites-Normal comparison
</summary>

File: shared/settings/invites/index.desktop.js [hunk 7]
```diff
@@ -33,12 +33,12 @@ function Invites (props: Props) {
         />}
         <Button
           type='Primary'
-          label='Generate invitiation'
```
Comment (chromakode) [inline shared/settings/invites/index.desktop.js]: Oops!

**Comment by keybase-ci-visdiff** [comment]
The commits f666fb7a535b...c60da233979f introduce visual changes on linux. <details><summary>:mag_right: 5 changed</summary>

 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/New_Proof_PGP_generating-Generating_PGP-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/New_Proof_PGP_generating-Generating_PGP-c60da233979f.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Empty_Message-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Empty_Message-c60da233979f.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Empty-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Empty-c60da233979f.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Email_Error-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal_Email_Error-c60da233979f.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475626033937-f666fb7a535b-c60da233979f-linux/Invites-Normal-c60da233979f.png)  
   Invites-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 3403b536a930...49573b1becb7 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Invites-Normal_Empty_Message** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Empty_Message-3403b536a930.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Empty_Message-49573b1becb7.png)  
   Invites-Normal_Empty_Message comparison
 * Changed: **Invites-Normal_Empty** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Empty-3403b536a930.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Empty-49573b1becb7.png)  
   Invites-Normal_Empty comparison
 * Changed: **Invites-Normal_Email_Error** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Email_Error-3403b536a930.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal_Email_Error-49573b1becb7.png)  
   Invites-Normal_Email_Error comparison
 * Changed: **Invites-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal-3403b536a930.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475679804529-3403b536a930-49573b1becb7-linux/Invites-Normal-49573b1becb7.png)  
   Invites-Normal comparison
</summary>

**Comment by cjb** [comment]
Ready for another look!  Adopted @chrisnojima's suggestions, and @MarcoPolo has pushed a commit adding subroute support to the settings nav, which was necessary to get the invite-generated component working.

File: shared/constants/settings.js [hunk 8]
```diff
@@ -23,7 +43,29 @@ export type DeleteAccountForever = NoErrorTypedAction<'settings:deleteAccountFor
 export type PlanLevel = 'Basic' | 'Gold' | 'Friend'
 const plans: Array<PlanLevel> = ['Basic', 'Gold', 'Friend']
 
-export type Actions = NotificationsRefresh | NotificationsRefreshed | NotificationsSave | NotificationsSaved | NotificationsToggle | SetAllowDeleteAccount
+export type PaymentInfo = {
```
Comment (MarcoPolo) [inline shared/constants/settings.js]: I moved this type out into constants/plan-billing. Do you need it here? or was this a bad merge?

File: shared/actions/settings.js [hunk 9]
```diff
@@ -66,6 +90,130 @@ function * saveNotificationsSaga (): SagaGenerator<any, any> {
...
+        uid: string,
+        username: string,
+      }],
+    } = JSON.parse(json && json.body || '')
+
+    const acceptedInvites = []
+    const pendingInvites = []
+
+    results.invitations.forEach(i => {
+      const invite: Constants.Invitation = {
```
Comment (MarcoPolo) [inline shared/actions/settings.js]: I don't think this type declaration doesn't work, you have to `import type {Invitation} from '...constants'`

File: shared/settings/dumb.js [hunk 10]
```diff
@@ -378,41 +378,38 @@ const invitesBase = {
...
       type: 'pending-url',
       id: '123457',
-      created: 1469566223000,
+      created: 1469566223,
       url: 'keybase.io/inv/9999999999',
     },
   ],
   acceptedInvites: [
     {
       id: '223456',
```
Comment (MarcoPolo) [inline shared/settings/dumb.js]: Don't these need uids? do they match the Invitation type?

File: shared/settings/nav/index.desktop.js [hunk 11]
```diff
@@ -34,7 +34,7 @@ class SettingsNav extends Component<void, Props, void> {
           <Box style={globalStyles.flexBoxColumn}>
             {items.map(i => <SettingsItem key={i.text} {...i} />)}
           </Box>
-          <Box style={{...globalStyles.flexBoxRow, flex: 1}}>
+          <Box style={{...globalStyles.flexBoxRow, flex: 1, overflow: 'auto'}}>
```
Comment (MarcoPolo) [inline shared/settings/nav/index.desktop.js]: Why was this change needed?

File: shared/settings/invites/index.desktop.js [hunk 12]
```diff
@@ -114,9 +114,9 @@ function PendingURLContent ({invite}: {invite: PendingURLInvite}) {
 function AcceptedInviteItem ({invite, onClick}: {invite: AcceptedInvite, onClick: (username: string) => void}) {
   const nameColor = stateColors(invite.currentlyFollowing, invite.trackerState, globalColors.blue).username
   return (
-    <Box style={{...styleInviteItem, ...globalStyles.clickable}} onClick={onClick}>
+    <Box style={{...styleInviteItem, ...globalStyles.clickable, flexShrink: 0}} onClick={onClick}>
       <Avatar
-        url={invite.avatar}
+        username={invite.username}
```
Comment (cjb) [inline shared/settings/invites/index.desktop.js]: Yeah, we don't get one now, just uid/username.  In the future we'll want to identify the uid we get back, so that we can find out whether they're followed and show that -- once we do that we'll have access to an avatar url.

**Comment by cjb** [comment]
@MarcoPolo Thanks for catching all of these!  I've made all the suggested changes, ready for another look.

I'm not sure I understood:
> OwnProps should be `Props - {onClose: () => void}`

I did:
```js
export type OwnProps = {
  onClose: () => void,
}

export default connect(
  (state: any, ownProps: OwnProps) => {
    return ownProps
  },
```
Is that what you were thinking?

**Comment by MarcoPolo** [comment]
👍
